framework/include/restart/DataIO.h

```diff
@@ -123,6 +126,13 @@ template<typename T>
 inline void
 dataStore(std::ostream & stream, T & v, void * /*context*/)
 {
+  #ifdef LIBMESH_HAVE_CXX11_TYPE_TRAITS
+    static_assert( std::is_polymorphic<T>::value == false,
+      "Cannot serialize a class that has virtual members!\nWrite a custom dataStore() template specialization!\n\n");
+    static_assert( std::is_trivially_copyable<T>::value || std::is_same<T, Point>::value,
```

Comment:
Oh yeah, although I merged this, I think we want to go back in and remove this second check since it's going to give us a false negative on classes which contain Points... unless we find some workaround for the non-trivial copy constructor issue.